sridha2412/assignment4/counting_islands.py

```diff
@@ -0,0 +1,48 @@
+from typing import List, Tuple
+import unittest
+
+class Grid:
+    def __init__(self, grid: List):
+        """ Constructor to initialize a grid """
+        self.grid = grid
+        self.row = len(self.grid)
+        self.col = len(self.grid[0])
+
+    def _get_neighbours(self, x: int, y: int) -> List[Tuple[int, int]]:
```

Comment:
To me it looks like get_neighbours could be simplified by merging it with  find_land. Another approach is to merge it  with __init__ and make a set of of all unvisited *land* position. Unvisited can then shrink as you explore each islands.